src/services/interfaces/IBaseServiceInterface.java

```diff
@@ -0,0 +1,16 @@
+package services.interfaces;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public interface IBaseServiceInterface<E, K> {
+  Optional<E> insert(E object);
+
```

Comment:
Why don't you change `object` by `entity` if you already have references to the word "entity" in your project?